python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1202,8 +1238,26 @@ def with_partition_keys(self, partition_keys: Iterable[str]) -> "TimeWindowParti
 
                 num_added_partitions += 1
 
+        return result_windows, num_added_partitions
+
+    def with_partition_keys(self, partition_keys: Iterable[str]) -> "TimeWindowPartitionsSubset":
+        # if we are representing things as a static set of keys, continue doing so
```

Comment:
This comments implies that keys vs. windows is either/or, but invoking `get_partition_keys` allows getting into a state where both are present, which I found a little confusing.